flow/core/rewards.py

```diff
@@ -330,16 +385,16 @@ def instantaneous_mpg(env, veh_ids=None, gain=.001):
         energy_model = env.k.vehicle.get_energy_model(veh_id)
         if energy_model != "":
             speed = env.k.vehicle.get_speed(veh_id)
-            accel = env.k.vehicle.get_accel_no_noise_with_failsafe(veh_id)
+            accel = env.k.vehicle.get_accel(veh_id, noise=False, failsafe=True)
             grade = env.k.vehicle.get_road_grade(veh_id)
             gallons_per_hr = energy_model.get_instantaneous_fuel_consumption(accel, speed, grade)
-            if gallons_per_hr > 0 and speed >= 0.0:
+            if speed >= 0.0:
                 cumulative_gallons += gallons_per_hr
                 cumulative_distance += speed
 
     cumulative_gallons /= 3600.0
     cumulative_distance /= 1609.0
     # miles / gallon is (distance_dot * \delta t) / (gallons_dot * \delta t)
-    mpg = cumulative_distance / cumulative_gallons
+    mpg = cumulative_distance / (cumulative_gallons + 1e-6)
```

Comment:
This would still blow up at really small values, no?

Reply:
this is just a stabilizing term for when a simulation starts. we shouldn't have this problem after the warmup steps are passed